### PR TITLE
PR Benchmark Diffing

### PR DESCRIPTION
**Pull request by MatthewCroughan**
This adds a new `effect` via https://github.com/hercules-ci/hercules-ci-effects which takes a pull request and performs a diff on the benchmark by using `nix run .#benchmark` on the PR `HEAD` and on the merge-base of that PR.

### CODE REVIEW

**Comment by L-as** [comment]
Please retarget to staging...

**Comment by L-as** [comment]
Is there any issue on GH for the bug?

**Comment by MatthewCroughan** [comment]
@L-as No, I'm in the process of making one.

**Comment by MatthewCroughan** [comment]
Turns out there was no bug, it was just a missing `)` in the `effectScript` which was easily conflated with the Nix stdenv when debugging.

File: flake.nix [hunk 1]
```diff
@@ -451,9 +456,43 @@
       );
       devShell = perSystem (system: self.flake.${system}.devShell);
 
-      nixCi = flake-compat-ci.lib.recurseIntoFlakeWith {
+      effects = { src }:
+        let
+          pkgs = nixpkgs.legacyPackages.x86_64-linux;
+          nix = (builtins.getFlake "github:matthewcroughan/nixcfg/f16df936ffc9850d42f68f4badef9b55d7e7fee8").nixosConfigurations.swordfish.pkgs.nixFlakes;
```
Comment (srid) [inline flake.nix:462]: Why do we need `github:matthewcroughan/nixcfg`? Isn't the nix executable available from the `pkgs` in line 461?

**Review by srid — APPROVED** [review]
Except for that one comment, we are good I think.

File: flake.nix (hunk 1 above)
Comment (MatthewCroughan) [reply flake.nix:462]: Ah, I left that in by mistake, this was something I was debugging.

**Pull request merged**